src/main/java/org/broadinstitute/hellbender/utils/text/ListFileUtils.java

```diff
@@ -0,0 +1,169 @@
+package org.broadinstitute.hellbender.utils.text;
+
+import htsjdk.samtools.util.IOUtil;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+/**
+ * A collection of convenience methods for working with list files.
+ */
+public class ListFileUtils {
+    /**
+     * Lines starting with this String in .list files are considered comments.
+     */
+    public static final String LIST_FILE_COMMENT_START = "#";        
+
+    /**
+     * Returns a new set of values, containing a final set of values expanded from values
+     * <p/>
+     * Each element E of values can either be a literal string or a file ending in .list.
+     * For each E ending in .list we try to read a file named E from disk, and if possible
+     * all lines from that file are expanded into unique values.
+     *
+     * @param values Original values
+     * @return entries from values or the files listed in values
+     */
+    public static Set<String> unpackSet(Collection<String> values) {
+        if (values == null)
+            throw new NullPointerException("values cannot be null");
+        Set<String> unpackedValues = new LinkedHashSet<String>();
+        // Let's first go through the list and see if we were given any files.
+        // We'll add every entry in the file to our set, and treat the entries as
+        // if they had been specified on the command line.
+        for (String value : values) {
+            File file = new File(value);
+            if (value.toLowerCase().endsWith(".list") && file.exists()) {
+                unpackedValues.addAll(new XReadLines(IOUtil.openFileForBufferedReading(file), true, LIST_FILE_COMMENT_START).readLines());
+            } else {
+                unpackedValues.add(value);
+            }
+        }
+        return unpackedValues;
+    }
```

Comment:
that's probably correct; however, do you have a GATK4 example of code using this barclay file expansion feature?

Reply:
Barclay does this automatically for any `@Argument` that is a collection (and that doesn't have suppressFileExpansion property set), so there isn't any code required if thats what you mean. SelectVariants, for example, relies on it - you can see an example of a test that uses it [here](https://github.com/broadinstitute/gatk/blob/f26cb5dbd7f8bb4ee549c7c1c1044e8ec7a28645/src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/SelectVariantsIntegrationTest.java#L72) if that helps.

Reply:
BTW, I noticed that SelectVariants has the filtering code from below embedded in it, since it does the same kind of thing. So that code should be updated to use whatever filtering methods result from these changes.

Reply:
That last post is exactly what I was going to write.  VariantEval is doing exactly what SelectVariants is doing, so I think we can entirely omit ListFileUtils.  Therefore we just want to pull that code related to SelectVariants.matchSamplesExpressions() into some kind of utility class.  Do you have an opinion on where in GATK4 this belongs?

Reply:
Probably Utils, assuming the code isn't sample-specific and is just filtering a list based on a regex.